pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -1070,7 +1244,32 @@ def _parse_control_values(control_wires, control_values):
         else:
             raise ValueError("Alternative control values must be passed as a binary string.")
 
-        return control_int
+        padding_left = control_int * 2
+        padding_right = 2 ** (len(control_wires)+1) - 2 - padding_left
+        cx = block_diag(np.eye(padding_left), PauliX.compute_matrix(), np.eye(padding_right))
+        return cx
+
+    def matrix(self, wire_order=None):
+
+        canonical_matrix = self.compute_matrix(self.hyperparameters["control_wires"],
+                                               self.hyperparameters["control_values"])
+
+        if self.inverse:
+            canonical_matrix = qml.math.conj(qml.math.T(canonical_matrix))
+
+        if wire_order is None or self.wires == Wires(wire_order):
+            return canonical_matrix
+
+        return qml.operation.expand_matrix(
+            canonical_matrix, wires=self.wires, wire_order=wire_order
+        )
+
+    @property
+    def control_wires(self):
+        return self.hyperparameters["control_wires"]
```

Comment:
I'm not sure we actually need this, I think it's for the circuit drawer. But in principle, the hyperparams now contain this information. Maybe delete in params/hyperparams refactor.

Reply:
I _think_ it makes sense to keep this if it is an important API for other parts of the codebase